File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/interfaceInJava/defaultImplementationInInterface/Car.java
package de.fraunhofer.iem.classesAndObjects.interfaceInJava.defaultImplementationInInterface;

/**
 * Demonstrates the implementing the interface that has the default implementation of a method
 *
 * @author Ranjith Krishnamurthy
 */
public class Car implements Vehicle {
    String modelNumber;
    String vehicleRegistrationNumber;
    private boolean isStart = false;

    public Car(String modelNumber, String vehicleRegistrationNumber) {
        this.modelNumber = modelNumber;
        this.vehicleRegistrationNumber = vehicleRegistrationNumber;
    }

    @Override
    public void displayModelNumber() {
        System.out.println(modelNumber);
    }

    @Override
    public void start() {
        if (isStart) {
            System.err.println("Car is already started.");
        }

        isStart = true;
    }

    @Override
    public void stop() {
        if (!isStart) {
            System.err.println("Car is already stopped.");
        }

        isStart = false;
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/interfaceInKotlin/defaultImplementationInInterface/Car.kt
package de.fraunhofer.iem.classesAndObjects.interfaceInKotlin.defaultImplementationInInterface

/**
 * Demonstrates the implementing the interface that has the default implementation of a method
 *
 * @author Ranjith Krishnamurthy
 */
class Car(override var modelNumber: String, override var vehicleRegistrationNumber: String) : Vehicle {
    private var isStart = false

    override fun displayModelNumber() {
        println(modelNumber)
    }

    override fun start() {
        if (isStart) {
            System.err.println("Car is already started.")
        }
        isStart = true
    }

    override fun stop() {
        if (!isStart) {
            System.err.println("Car is already stopped.")
        }
        isStart = false
    }
}